### PR TITLE
Enabled type-checking in VSCode and fix found issues

### PR DESCRIPTION
**Pull request by AlCalzone**
Same as I already did with zwave and another adapter I don't remember right now. Many "issues" could be fixed by asserting the variable type myself.

A bunch of the reported issues were with `fs.readFileSync()` which expects the encoding as the 2nd parameter. The fallback to `utf8` if none is given is not documented, so we should avoid it.

Aside from that, there were some actual bugs (undefined variables). I commented below where I found them.

### CODE REVIEW

File: lib/setup/setupVisDebug.js [hunk 1]
```diff
@@ -91,13 +91,13 @@ function VisDebug(options) {
         objects.writeFile('vis', 'edit.html', file);
 
         console.log('Modify "' + path.normalize(visDir + '/www/cache.manifest') + '"');
-        file = fs.readFileSync(visDir + '/www/cache.manifest').toString();
+        file = fs.readFileSync(visDir + '/www/cache.manifest', 'utf8');
         var n = file.match(/# dev build (\d+)/, '5');
```
Comment (AlCalzone) [inline lib/setup/setupVisDebug.js]: VSCode complains that `match` only takes one argument. What does the '5' do?

File: lib/setup/setupRepo.js [hunk 2]
```diff
@@ -99,7 +99,7 @@ function Repo(options) {
                 text += '"' + name + '"' + ((name.length < 15) ? new Array(15 - name.length).join(' ') : '');
 
                 var tLen = 10;
-                if (name.length >= 15) tLen -= (name.length > tLen);
+                if (name.length >= 15) tLen -= (name.length > tLen ? 1 : 0);
```
Comment (AlCalzone) [inline lib/setup/setupRepo.js]: This syntax was unclear as we subtract a boolean from a number.

File: lib/setup/setupRepo.js [hunk 3]
```diff
@@ -72,7 +72,7 @@ function Repo(options) {
             downloads.push(name);
         }
 
-        download(sources);
+        download(/*sources*/);
```
Comment (AlCalzone) [inline lib/setup/setupRepo.js]: `download` takes no arguments and sources is captured by the closure anyways.

File: lib/restart.js [hunk 4]
```diff
@@ -161,7 +161,7 @@ if (require('os').platform().match(/^win/) && fs.existsSync(__dirname + '/../_se
                     });
                 });
             } catch (e) {
-                log('Error by pids.txt', e);
+                log('Error by pids.txt: ' + e);
```
Comment (AlCalzone) [inline lib/restart.js]: `log` only takes 1 argument

File: lib/objects/objectsInRedis.js [hunk 5]
```diff
@@ -1458,6 +1461,7 @@ function ObjectsInMem(settings) {
                 }
                 if (obj && obj.views && obj.views[search]) {
                     objects[obj._id] = obj;
+                    // ERROR: result not found
                     that._applyView(result, obj.views[search], params, options, callback);
```
Comment (AlCalzone) [inline lib/objects/objectsInRedis.js]: Where's `result` defined?

File: lib/objects/objectsInRedis.js [hunk 6]
```diff
@@ -535,6 +537,7 @@ function ObjectsInMem(settings) {
             if (_obj) {
                 _obj._attachments = _obj._attachments || {};
                 if (!_obj._attachments[name]) {
+                    // ERROR: mimeType not found
                     _obj._attachments[name] = mimeType;
```
Comment (AlCalzone) [inline lib/objects/objectsInRedis.js]: where's `mimeType` defined?

File: lib/objects/objectsInRedis.js [hunk 7]
```diff
@@ -170,9 +171,10 @@ function ObjectsInMem(settings) {
 
         // read rights of file
         if (!fileOptions[id]) {
+            // ERROR: objectsDir not found!
             if (fs.existsSync(objectsDir + id + '/_data.json')) {
```
Comment (AlCalzone) [inline lib/objects/objectsInRedis.js]: where's `objectsDir` defined?

File: lib/setup/setupVisDebug.js (hunk 1 above)
Comment (Apollon77) [reply lib/setup/setupVisDebug.js]: Good question, I would say we remove it?!

File: lib/setup/setupRepo.js (hunk 2 above)
Comment (Apollon77) [reply lib/setup/setupRepo.js]: makes sense

**Comment by Apollon77** [comment]
You found interesting things :-)
Bluefox should check them

**Comment by AlCalzone** [comment]
@GermanBluefox what about the "not found" errors? Don't forget them ;)

**Pull request merged**